test/e2e/e2e_test.go

```diff
@@ -464,6 +472,127 @@ var _ = ginkgo.Describe("Kueue", func() {
+			lookupKey1 := types.NamespacedName{Name: sampleJob.Name, Namespace: sampleJob.Namespace}
+			wll := &kueue.Workload{}
+
+			ginkgo.By("checking the job starts")
+			gomega.Expect(k8sClient.Create(ctx, sampleJob)).Should(gomega.Succeed())
+
+			createdJob := &batchv1.Job{}
+			wlKey := types.NamespacedName{Name: workloadjob.GetWorkloadNameForJob(sampleJob.Name), Namespace: sampleJob.Namespace}
+
+			gomega.Eventually(func(g gomega.Gomega) {
+				g.Expect(k8sClient.Get(ctx, lookupKey1, sampleJob)).Should(gomega.Succeed())
+				g.Expect(sampleJob.Spec.Suspend).To(gomega.Equal(ptr.To(false)))
+				g.Expect(k8sClient.Get(ctx, wlKey, wll)).Should(gomega.Succeed())
+				util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, wll)
+			}, util.Timeout, util.Interval).Should(gomega.Succeed())
+
+			ginkgo.By("Change the Active field to suspend the job and check the job remains suspended and the workload unadmitted")
+			// Changing QueuingPolicy to Never
+			wll.Spec.Active = false
+			gomega.Expect(k8sClient.Update(ctx, wll)).Should(gomega.Succeed())
+
+			ginkgo.By("checking a second job starts after first job is suspended")
+			sampleJob2 = (&testingjob.JobWrapper{Job: *sampleJob2}).Image("gcr.io/k8s-staging-perf-tests/sleep:v0.0.3", []string{"1s"}).Obj()
+			wll2 := &kueue.Workload{}
+
+			gomega.Expect(k8sClient.Create(ctx, sampleJob2)).Should(gomega.Succeed())
+			wlKey2 := types.NamespacedName{Name: workloadjob.GetWorkloadNameForJob(sampleJob2.Name), Namespace: sampleJob2.Namespace}
+
+			gomega.Eventually(func(g gomega.Gomega) {
+				g.Expect(k8sClient.Get(ctx, jobKey2, sampleJob2)).Should(gomega.Succeed())
+				g.Expect(sampleJob2.Spec.Suspend).To(gomega.Equal(ptr.To(false)))
+				g.Expect(k8sClient.Get(ctx, wlKey2, wll2)).Should(gomega.Succeed())
+				util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, wll2)
+			}, util.Timeout, util.Interval).Should(gomega.Succeed())
+
+			// Checking job stays suspended
+			ginkgo.By("checking job stays suspended")
+			gomega.Eventually(func() *bool {
+				gomega.Expect(k8sClient.Get(ctx, types.NamespacedName{Name: sampleJob.Name, Namespace: sampleJob.Namespace}, createdJob)).
+					Should(gomega.Succeed())
+				return createdJob.Spec.Suspend
+			}, util.Timeout, util.Interval).Should(gomega.Equal(ptr.To(true)))
+
+			// Job should stay pending
+			gomega.Consistently(func() *bool {
+				gomega.Expect(k8sClient.Get(ctx, types.NamespacedName{Name: sampleJob.Name, Namespace: sampleJob.Namespace}, createdJob)).
+					Should(gomega.Succeed())
+				return createdJob.Spec.Suspend
+			}, util.ConsistentDuration, util.Interval).Should(gomega.Equal(ptr.To(true)))
+
+			ginkgo.By("checking the first workload gets unadmitted")
+			// Workload should get unadmitted
+			gomega.Expect(k8sClient.Get(ctx, wlKey, wll)).Should(gomega.Succeed())
+			util.ExpectWorkloadsToBePending(ctx, k8sClient, wll)
+
+			// Workload should stay pending
+			gomega.Consistently(func() bool {
```

Comment:
You have 2 "Consistently" operations. This adds too much runtime to the test. You can combine both checks into a single "Consistently".